src/main/kotlin/fsrss/FeedController.kt

```diff
@@ -1,17 +1,63 @@
 package fsrss
 
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Controller
-import org.springframework.web.bind.annotation.GetMapping
-import org.springframework.web.bind.annotation.PathVariable
-import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.client.HttpClientErrorException
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
 
 
 @Controller
 class FeedController {
+    val path: String = System.getenv("RSS_FEEDS_PATH") ?: "/"
+    val jsonMapper = jacksonObjectMapper()
 
     @GetMapping("/feeds/{feed}")
     @ResponseBody
-    fun readFeed(@PathVariable feed: String) :  String {
-        return "<rss>$feed</rss>"
+    fun readFeed(@PathVariable feed: String) : String {
+        val path = Path.of(path, feed)
+        if (Files.exists(path)) {
+            return FeedHandler().createRssFeed(path.toFile())
+        } else {
+            throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Feed does not exist")
+        }
+    }
+
+    @RequestMapping("/feeds/{feed}")
+    @ResponseStatus(HttpStatus.OK)
+    fun createFeed(@PathVariable feed: String, @RequestBody body: String) {
+        val path = Path.of(path + feed)
+        if (!Files.exists(path)) {
+            val meta: FeedMetaData = jsonMapper.readValue(body)
+            Files.createDirectory(path)
+            val metaFile = File("$path/meta.json")
+            metaFile.createNewFile()
+            metaFile.writeText(jsonMapper.writeValueAsString(meta))
+        } else {
+            throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Feed already exist")
+        }
+    }
+
+    @RequestMapping("/feeds/{feed}/item")
```

Comment:
We could probably use the "channel" level as well, so /feeds/{feed}/channels/{channel}/items/{item}

Reply:
There can only by one channel inside a RSS feed as far as I know. There is nothing specific to them